app/models/transformation_mapping.rb

```diff
@@ -0,0 +1,67 @@
+class TransformationMapping < ApplicationRecord
+  has_many :transformation_mapping_items, :dependent => :destroy
+
+  validates :name, :presence => true, :uniqueness => true
+
+  virtual_column :mappings, :type => :string
+
+  # forward compatible: next release the table will be polymorphic
+  virtual_column :type,     :type => :string
+
+  def self.inheritance_column
+    "none"
+  end
+
+  def type
+    "MigrationInfrastructureMapping"
+  end
+
+  def type=(type)
+  end
+
+  # Assumption: source and destination types are the same. Works for infrastructure mapping
+  # group mapping items first by destination_type, then by destination_id
+  # simplify records with IDs.
+  # Sample result (Hash)
+  #
+  # EmsCluster:
+  #   - destination: "2":
+  #     sources: ["100", "200"]
+  #   - destination: "3"
+  #     sources: ["301", "302"]
+  # Network:
+  #   - destination: "2"
+  #     sources: ["202", "204"]
+  # Storage:
+  #   - destination: "4"
+  #     sources: ["33", "32"]
+  def mappings
+    dst_type_groups = transformation_mapping_items.group_by(&:destination_type)
+    dst_type_groups.each_with_object({}) do |(dst_type, items), results|
+      dst_groups = items.group_by { |item| item.destination_id.to_s }
+      results[dst_type] = dst_groups.each_with_object([]) do |(dst_id, item_array), dst_id_groups|
+        dst_id_groups << {"destination" => dst_id, "sources" => item_array.collect { |item| item.source_id.to_s }}
+      end
+    end
+  end
+
+  def mappings=(options)
+    options.each do |dst_type, item_map_array|
+      item_map_array.each do |item_map|
+        destination_id = item_map['destination']
+        item_map['sources'].each do |source_id|
+          transformation_mapping_items << TransformationMappingItem.new(
+            :source_id        => source_id,
+            :source_type      => dst_type,
```

Comment:
Just had an idea. Instead of using the `dst_type`, how do you feel about the API sending you a `source_type`? Something along the lines of:
```
{"name": "new transformation mapping",
 "mappings": { 
    "clusters": [{"source_type": "EmsCluster", "destination": ":id", "sources": [":id"]}
  ]}}
```
We would add this in on the `create` (instead of just changing `cluster` to `EmsCluster`), and would also mean that we would not need to do the reverse on a `GET` (`EmsCluster` to `cluster`)